src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -104,6 +107,11 @@ public static class TagHelperBlockRewriter
 
                 if (afterEquals)
                 {
+                    // We've captured all leading whitespace, the attribute name, and an equals with an optional 
+                    // quote/double quote. We're now at: " asp-for='|...'" or " asp-for=|..."
+                    // The goal here is to capture all symbols until the end of the attribute. Note this will not 
+                    // consume an ending quote due to the symbolOffset.
```

Comment:
teensy nit: a second note that whitespace between the attribute value or the closing quote and the next attribute or the end of the tag is not included in `htmlSymbols` may be worthwhile.